torch/_inductor/fx_passes/mkldnn_fusion.py

```diff
@@ -0,0 +1,260 @@
+            if isinstance(min_value, ir.TensorBox) or isinstance(
+                max_value, ir.TensorBox
+            ):
+                matched = False
+            else:  # inp is a Number
+                matched = True
+            computation_args = list(args)
+            if matched:
+                computation_args = computation_args[:-3] + [
+                    "hardtanh",
+                    [min_value, max_value],
+                    "",
+                ]
+                return L[computation_op](*computation_args)
+            else:
+                conv_out = L[computation_op](*computation_args)
+                return L[aten.clamp_max](
+                    L[aten.clamp_min](conv_out, min_value), max_value
+                )
+
+        return fn
+
+    def _register_unary_fusion():
+        class UnaryAttr:
+            def __init__(self, op_name: str, scalars_attr=None, algorithm_attr=None):
+                self.op_name = op_name
+                self.scalars_attr = scalars_attr if scalars_attr else []
+                self.algorithm_attr = algorithm_attr if algorithm_attr else ""
+
+        replacement_unary_fusion_patterns = {
+            UnaryAttr("gelu", algorithm_attr="tanh"): [
+                _gelu_fusion_2(u) for u in _computation_user_4
+            ],
+            UnaryAttr("gelu", algorithm_attr="none"): [
+                _gelu_fusion_1(u) for u in _computation_user_2
+            ],
+            UnaryAttr("hardswish"): [_hardswish_fusion(u) for u in _computation_user_2],
+            UnaryAttr("hardsigmoid"): [
+                _hardsigmoid_fusion(u) for u in _computation_user_1
+            ],
+            UnaryAttr("swish"): [_silu_fusion(u) for u in _computation_user_2],
+            UnaryAttr("relu"): [
+                _combined_fusion(u, aten.relu) for u in _computation_user_1
+            ],
+            UnaryAttr("sigmoid"): [
+                _combined_fusion(u, aten.sigmoid) for u in _computation_user_1
+            ],
+            UnaryAttr("tanh"): [
+                _combined_fusion(u, aten.tanh) for u in _computation_user_1
+            ],
+        }
+        computation_ops = [mkldnn._convolution_pointwise.default]
+
+        for unary_attr, patterns in replacement_unary_fusion_patterns.items():
+            _register_unary_fusion_lowering(patterns[0], unary_attr, computation_ops[0])
+            # TODO: add linear/ConvTranspose fusion
+
+        _leaky_relu_patterns = [
+            _leaky_relu_fusion(user) for user in _computation_user_3
+        ]
+        _hardtanh_patterns = [_hardtanh_fusion(user) for user in _computation_user_1]
+        for pattern, computation_op in zip(_leaky_relu_patterns, computation_ops):
+            _register_leaky_relu_fusion_lowering(pattern, computation_op)
+        for pattern, computation_op in zip(_hardtanh_patterns, computation_ops):
+            _register_hardtanh_fusion_lowering(pattern, computation_op)
+
+    def _mkldnn_fusion_init():
```

Comment:
```suggestion
    @functools.lru_cache(None)
    def _mkldnn_fusion_init():
```
So this code only runs once.

Reply:
changed.